csrank/tests/test_ranking.py

```diff
@@ -1,15 +1,36 @@
 import logging
-import pytest
+import os
 from abc import ABCMeta
 
 import numpy as np
+import pytest
+import tensorflow as tf
 from keras.layers import Input
 from keras.models import Model
 from keras.optimizers import SGD
 from keras.regularizers import l2
 
+from csrank import FETANetwork, RankNet, CmpNet, ExpectedRankRegression, RankSVM
 from ..fate_ranking import FATERankingCore, FATEObjectRanker
-from ..util import tunable_parameters_ranges
+from ..util import tunable_parameters_ranges, zero_one_rank_loss_for_scores_ties_np
+
+RANKSVM = 'ranksvm'
+ERR = 'err'
+CMPNET = "cmpnet"
+RANKNET = 'ranknet'
+FETA_RANKER = 'feta_ranker'
+FATE_RANKER = "fate_ranker"
+
+object_rankers = {FETA_RANKER: FETANetwork, RANKNET: RankNet, CMPNET: CmpNet,
+                  ERR: ExpectedRankRegression, RANKSVM: RankSVM,
+                  FATE_RANKER: FATEObjectRanker}
+object_rankers_params = {
+    FETA_RANKER: {"add_zeroth_order_model": True, "optimizer": SGD(lr=1e-3, momentum=0.9, nesterov=True)},
+    RANKNET: {"optimizer": SGD(lr=1e-3, momentum=0.9, nesterov=True)},
+    CMPNET: {"optimizer": SGD(lr=1e-3, momentum=0.9, nesterov=True)},
```

Comment:
This looks like we can factor SGD out and create it in advance.

Reply:
Done